Developer Release Notes documentation

sf1919 opened the pull request:
**Description of work.**
To enable developers to start using the new release notes system this documentation has been added to assist. It covers where to save release notes, file naming conventions and how to expand the headings available as standard. 

The Github PR template is also updated to provide relevant guidance to developers.

Please note: this is blocked by #33577 as any changes to folder names may need to be reflected in this documentation.

**To test:**
Read through and check for
- clarity
- consistency
- spelling & grammar
- working links
- correct formatting

*There is no associated issue.*


*This does not require release notes* because **it is developer guidance.**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Updated PR template

Added new directions to Github PR template for adding release notes and reviewing release notes.

sf1919 committed:
Added Release Note Guide

- Added a guide for developers for release notes
- Included link in Standards index

sf1919 committed:
Updated file naming conventions

- revised the list of file naming conventions

MialLewis reviewed (commented):
Clear and concise documentation. A few very minor suggestions/typo correction for your consideration.

sf1919 commented on dev-docs/source/Standards/ReleaseNotesGuide.rst:
```diff
@@ -0,0 +1,232 @@
+There will be occasions when the range of standard headings is not suitable for your needs. If you want to add a heading you need to do the following:-
+
+- Add a new directory into the correct part of the filing system. E.g. to add Algorithms to Workbench create the directory ``/Workbench/Algorithm``. The directory name should not contain any spaces.
+- Add ``New_features``, ``Improvements`` and ``Bugfixes`` directories within your new directory. The automated script only works with these directories.
```
You can add sub directories but those directories must contain the New_features, Improvements and Bugfixes directories within it. The script only works if the final layer of directories is these 3 on any given branch. I've tried to convey that in this section but maybe thats not clear?

MialLewis replied:
That was clear, I wondered if you could add sub-directories to the New_features, Improvements and Bugfixes sub-directories (again, not sure if this would ever be necessary).

sf1919 replied:
Any sub directories added to New_features or Bugfixes would not be picked up by the script so these release notes would be effectively invisible. 

Having reviewed this I'm removing Improvements and leaving it as New_features and Bugfixes as it is much simpler going forward (after all what is the difference between a new feature and an improvement?).

MialLewis replied:
I think this is the only section that isn't explicitly clear in my mind. Perhaps due to the terms "heading" and "sub-heading" being used interchangeably?

Perhaps make it clear that 
```
* Diffraction **<-This is a heading**

  - Powder Diffraction **<-This is a sub-heading**

	  + New features
	  + Bugfixes

  - Engineering Diffraction

	  + New features
	  + Bugfixes

  - Powder Diffraction

	  + New features
	  + Bugfixes
```

You **cannot** add new headings, you **can** add new sub-headings, you **cannot** add further layers of headings to extend this structure (eg. sub-sub headings).

If I've interpreted that correctly?

sf1919 commented on dev-docs/source/Standards/ReleaseNotesGuide.rst:
```diff
@@ -0,0 +1,232 @@
+
+Adding release notes
+--------------------
+Release notes are added as new files in a pull request to avoid merge conflicts. Automated scripts then amalgamate these into the release notes available for users. If your work requires more than one note for the same subheading then it is fine to put them all in the same file. However,
```
This hasn't been decided yet but once it has I expect we'll update the documentation with that information.

sf1919 committed:
Improvements to Release Note Guide

- Removed references to Improvements as going forward all release notes will either be New features or Bugfixes
- Added new section on previewing release notes
- removed spelling mistakes spotted in review

sf1919 commented:
Part of #33593

MialLewis commented on dev-docs/source/Standards/ReleaseNotesGuide.rst:
```diff
@@ -0,0 +1,215 @@
+There are two ways you can preview release notes prior to them being merged in.
+
+1. Build the documents and view the release note from the build folder as a HTML file. You will need to find it in the directory structure and it may take some time if you have not recently built documentation.
+2. Via the Pull Request (PR). You can view the release note on Github and it will show it using basic .rst rendering. You cannot check all the features you might expect to see when the release note is merged in (e.g. you cannot
```
It looks to me like the ident might be off on this bullet, I've checked out the branch but can't see how to view the dev docs to check properly.

sf1919 committed:
Revisions on review

- reformatted numbered list
- tried to clarify main headings and sub-headings and what can be added where.

DavidFair reviewed (approved):
LGTM, looking forward to these changes :+1:
